odoo/addons/base/models/ir_ui_view.py

```diff
@@ -41,6 +41,26 @@
 MOVABLE_BRANDING = ['data-oe-model', 'data-oe-id', 'data-oe-field', 'data-oe-xpath', 'data-oe-source-id']
 
 
+def quick_eval(expr, globals_dict):
+    """ Functionally identical to safe_eval(), but optimized with special-casing. """
+    # most (~95%) elements are 1/True/0/False
+    if expr == '1':
+        return 1
+    if expr == 'True':
+        return True
+    if expr == '0':
+        return 0
+    if expr == 'False':
+        return False
+    return safe_eval.safe_eval(expr, globals_dict)
```

Comment:
maybe add this in safe_eval()

Reply:
The other places where we use `safe_eval()` generally don't expect those values.  Those values are very common in views, but not at all in server actions or `ir.rule` domains, for instance.  That's why I think this optimization is more specific to views, and should remain specific.